libgalois/include/katana/MemoryManager.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include <memory>
+#include <unordered_map>
+
+#include "katana/Manager.h"
+#include "katana/Result.h"
+#include "katana/config.h"
+
+namespace katana {
+// The memory manager singleton (MM).  Not thread safe.
+// The MM controls policy and does bookkeeping.  All memory allocation
+// is done by the system, mostly the C++ standard library.
+// The MM does not manage per-allocation tokens, it only manages sizes
+// Clients must call the proper functions or the MM will make
+// bad decisions.
+
+class KATANA_EXPORT MemoryManager {
+public:
+  MemoryManager(const MemoryManager&) = delete;
+  MemoryManager(MemoryManager&&) = delete;
+  MemoryManager& operator=(const MemoryManager&) = delete;
+  MemoryManager& operator=(MemoryManager&&) = delete;
+
+  // https://codereview.stackexchange.com/questions/173929/modern-c-singleton-template
+  static auto& MM() {
```

Comment:
Why not just `Get()`? I'll probably make `getSystemNetworkInterface()` a static member `Get()` of `NetworkInterface`

Reply:
I was being a bit cutesy.  I like `MM `as the name of the memory manager singleton, but `Get `is probably more uniform.